ui/index.js

```diff
@@ -104,3 +101,34 @@ async function startApp (metamaskState, backgroundConnection, opts) {
 
   return store
 }
+
+function setupDebuggingHelpers (store) {
+  window.getCleanAppState = function () {
+    const state = clone(store.getState())
+    state.version = global.platform.getVersion()
+    state.browser = window.navigator.userAgent
+    return state
+  }
+}
+
+window.logStateString = function (cb) {
+  const state = window.getCleanAppState()
+  global.platform.getPlatformInfo((err, platform) => {
+    if (err) {
+      return cb(err)
+    }
+    state.platform = platform
+    const stateString = JSON.stringify(state, null, 2)
+    cb(null, stateString)
+  })
+}
+
+window.logState = function () {
+  return window.logStateString((err, result) => {
```

Comment:
This seems like a bad function in general - it's doing an `async` thing, but it neither accepts a callback nor returns a promise. Worse, it also returns a value.

What was wrong with the way it was before, with the boolean parameter for copying to the clipboard?

Reply:
Sure, I'll leave it as it was

Reply:
> Worse, it also returns a value.

Yeah and the callback from `logStateString` doesn't return _its_ callback value, so... yeah. My bad.